challenge/tlsalpn01/tls_alpn_challenge_test.go

```diff
@@ -116,3 +119,75 @@ func TestChallengeInvalidPort(t *testing.T) {
 	assert.Contains(t, err.Error(), "invalid port")
 	assert.Contains(t, err.Error(), "123456")
 }
+
+func TestChallengeIPaddress(t *testing.T) {
+	_, apiURL := tester.SetupFakeAPI(t)
+
+	domain := "127.0.0.1"
+	port := "23457"
+
+	mockValidate := func(_ *api.Core, _ string, chlng acme.Challenge) error {
+		conn, err := tls.Dial("tcp", net.JoinHostPort(domain, port), &tls.Config{
+			InsecureSkipVerify: true,
+		})
+		require.NoError(t, err, "Expected to connect to challenge server without an error")
+
+		// Expect the server to only return one certificate
+		connState := conn.ConnectionState()
+		assert.Len(t, connState.PeerCertificates, 1, "Expected the challenge server to return exactly one certificate")
+
+		remoteCert := connState.PeerCertificates[0]
+		assert.Len(t, remoteCert.DNSNames, 0, "Expected the challenge certificate to have no DNSNames entry in context of challenge for IP")
+		assert.Len(t, remoteCert.IPAddresses, 1, "Expected the challenge certificate to have exactly one IPAddresses entry")
+		assert.True(t, net.ParseIP("127.0.0.1").Equal(remoteCert.IPAddresses[0]), "challenge certificate IPAddress ")
+		assert.NotEmpty(t, remoteCert.Extensions, "Expected the challenge certificate to contain extensions")
+
+		idx := -1
+		for i, ext := range remoteCert.Extensions {
+			if idPeAcmeIdentifierV1.Equal(ext.Id) {
+				idx = i
+				break
+			}
+		}
+
+		require.NotEqual(t, -1, idx, "Expected the challenge certificate to contain an extension with the id-pe-acmeIdentifier id,")
+
+		ext := remoteCert.Extensions[idx]
+		assert.True(t, ext.Critical, "Expected the challenge certificate id-pe-acmeIdentifier extension to be marked as critical")
+
+		zBytes := sha256.Sum256([]byte(chlng.KeyAuthorization))
+		value, err := asn1.Marshal(zBytes[:sha256.Size])
+		require.NoError(t, err, "Expected marshaling of the keyAuth to return no error")
+
+		if subtle.ConstantTimeCompare(value, ext.Value) != 1 {
+			t.Errorf("Expected the challenge certificate id-pe-acmeIdentifier extension to contain the SHA-256 digest of the keyAuth, %v, but was %v", zBytes[:], ext.Value)
+		}
```

Comment:
`require.EqualValues` should suffice here:

```suggestion
		require.EqualValues(t, value, ext.Value, "Expected the challenge certificate id-pe-acmeIdentifier extension to contain the SHA-256 digest of the keyAuth")
```

(I may have the order of expected/actual value wrong)

Reply:
it's copied from DNS name version, do you want to change it in this commit too?

Reply:
Oh, I didn't realize that.

It would be nice if you could cleanup the other test version as well, yes.